src/wrapper/java_vm/vm.rs

```diff
@@ -154,20 +154,22 @@ impl JavaVM {
         }
     }
 
-    /// Detaches current thread from the JVM.
+    /// Detaches current thread from the JVM. This operation is _rarely_ appropriate to use,
+    /// because the attachment methods [ensure](#attaching-native-threads) that the thread is
+    /// promptly detached.
     ///
-    /// Detaching a non-attached thread is no-op.
+    /// Detaching a non-attached thread is a no-op.
     ///
-    /// Calling this method is an equivalent for calling `drop()` for `AttachGuard`. However, the
-    /// actual guard will not be dropped. __Any existing `JNIEnv` and `AttachGuard` will be
-    /// considered invalid in the current thread after calling this method.__ Any operation
-    /// on `JNIEnv` will return an error.
+    /// __Any existing `JNIEnv`s and `AttachGuard`s created in the calling thread
+    /// will be invalidated after this method completes. It is the__ caller’s __responsibility
```

Comment:
I wonder if this will render correctly, it seems like underscores on this line are redundant

Reply:
Oh well, it will not render correctly, because it will not render at all :D
We hide this method, so it seems we should not use markdown formatting here, as it is useless

Reply:
They are not redundant — they highlight caller’s from the rest of the text, which is in bold. You will see highlighting if you use "Show quick documentation" in your IDEA (E.g., Ctrl + Q).

Reply:
Okay, I returned them back